pkg/cmd/cli/cmd/newapp.go

```diff
@@ -532,6 +532,10 @@ func CompleteAppConfig(config *newcmd.AppConfig, f *clientcmd.Factory, c *cobra.
 	if config.BinaryBuild && config.Strategy == generate.StrategyPipeline {
 		return kcmdutil.UsageError(c, "specifying binary builds and the pipeline strategy at the same time is not allowed.")
 	}
+
+	if len(config.BuildArgs) > 0 && config.Strategy != generate.StrategyDocker {
+		return kcmdutil.UsageError(c, "build arguments can only be used with Docker builds.")
+	}
```

Comment:
do we necessarily know the strategy that's going to be used at this point?  what is the behavior when no strategy is explicitly specified?